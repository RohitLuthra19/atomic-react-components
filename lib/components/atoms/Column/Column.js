// @flow
/**
 *
 * Column
 *
 */
import React from 'react';
import styled from 'styled-components';

import styles from './Column.style';
import type { Props } from './types';

const Column = ({ intent, label, value }) =>
  intent === 'header' ? <span>{label}</span> : <span>{value}</span>;

Column.defaultProps = {};

export default styled(Column)`
  ${styles};
`;

export { Column as ColumnVanilla };
