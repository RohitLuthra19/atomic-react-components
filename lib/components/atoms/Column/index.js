// @flow
export { default } from './Column';
export { ColumnVanilla } from './Column';
