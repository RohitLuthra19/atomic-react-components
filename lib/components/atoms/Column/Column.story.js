import React from 'react';
import { storiesOf } from '@storybook/react';
// import { action } from '@storybook/addon-actions';
import defaultConfig from './Column.mock';

// Import Styled Component to showcase variations
import Column, { ColumnVanilla } from '.';

// TODO: indentation is not working fine in editor. Need to check for .hbs files
// $FlowFixMe eslint-disable-next-line extra-rules/no-commented-out-code
storiesOf('Atoms', module)
  .addParameters({ jest: ['Column', 'ColumnVanilla'] })
  .add('CTA Knobs', () => (
    <ColumnVanilla
      {...defaultConfig}
      className={`hide-default-sample ${defaultConfig.className}`}
    />
  ))
  .add('CTA', () => <Column {...defaultConfig} />);
