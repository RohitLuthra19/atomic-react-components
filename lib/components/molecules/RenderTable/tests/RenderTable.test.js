import { shallow } from 'enzyme';

import { RenderTableVanilla } from '../index';
import Column from '../../../atoms/Column';

describe('RenderTable', () => {
  let wrapper;
  let table;
  let thead;
  let headers;
  let tbody;
  let rows;
  const data = [
    { name: 'Cupcake', calories: 305, fat: 3.7, carbs: 67, protein: 4.3, id: '1' },
    { name: 'Donut', calories: 452, fat: 25, carbs: 51, protein: 4.9, id: '2' },
    { name: 'Eclair', calories: 262, fat: 16, carbs: 24, protein: 6, id: '3' },
    { name: 'Frozen yoghurt', calories: 159, fat: 6, carbs: 24, protein: 4, id: '5' },
    { name: 'Gingerbread', calories: 356, fat: 16, carbs: 49, protein: 3.9, id: '6' },
    { name: 'Honeycomb', calories: 408, fat: 3.2, carbs: 87, protein: 6.5, id: '7' },
    { name: 'Ice cream sandwich', calories: 237, fat: 9, carbs: 37, protein: 4.3, id: '8' },
    { name: 'Jelly Bean', calories: 375, fat: 0, carbs: 94, protein: 0, id: '9' },
    { name: 'KitKat', calories: 518, fat: 26, carbs: 65, protein: 7, id: '10' },
    { name: 'Lollipop', calories: 392, fat: 0.2, carbs: 98, protein: 0, id: '11' },
    { name: 'Marshmallow', calories: 318, fat: 0, carbs: 81, protein: 2, id: '12' },
    { name: 'Nougat', calories: 360, fat: 19, carbs: 9, protein: 37, id: '13' },
    { name: 'Oreo', calories: 437, fat: 18, carbs: 63, protein: 4, id: '14' },
  ];
  beforeEach(() => {
    wrapper = shallow(
      <RenderTableVanilla data={data}>
        <Column label="Dessert (100g serving)" name="name" />
        <Column label="Calories" name="calories" />
        <Column label="Fat (g)" name="fat" />
        <Column label="Carbs (g)" name="carbs" />
        <Column label="Protein (g)" name="protein" />
      </RenderTableVanilla>
    );
    table = wrapper.find('table');
    thead = table.find('thead');
    tbody = table.find('tbody');
    headers = thead.find('th');
    rows = tbody.find('tr');
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have table component', () => {
    expect(table.length).toEqual(1);
  });

  it('should have thead component', () => {
    expect(thead.length).toEqual(1);
  });

  it('should have tbody component', () => {
    expect(tbody.length).toEqual(1);
  });

  it('should have th is equal to number of columns', () => {
    expect(headers.length).toEqual(5);
  });

  it('should have tr is equal to number of data rows', () => {
    expect(rows).toHaveLength(data.length);
  });

  it('should loop through each row and check the content', () => {
    rows.forEach((tr, rowIndex) => {
      const cells = tr.find('td');
      expect(cells).toHaveLength(5);
      expect(cells.at(0).children().prop('value')).toEqual(data[rowIndex].name);
      expect(cells.at(1).children().prop('value')).toEqual(data[rowIndex].calories);
      expect(cells.at(2).children().prop('value')).toEqual(data[rowIndex].fat);
      expect(cells.at(3).children().prop('value')).toEqual(data[rowIndex].carbs);
      expect(cells.at(4).children().prop('value')).toEqual(data[rowIndex].protein);
    });
  });
});
