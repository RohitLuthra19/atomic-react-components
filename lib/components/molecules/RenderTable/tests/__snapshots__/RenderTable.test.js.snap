// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderTable should render correctly 1`] = `
<table>
  <thead>
    <tr>
      <th
        key="name/.0"
      >
        <Column
          intent="header"
          label="Dessert (100g serving)"
          name="name"
        />
      </th>
      <th
        key="calories/.1"
      >
        <Column
          intent="header"
          label="Calories"
          name="calories"
        />
      </th>
      <th
        key="fat/.2"
      >
        <Column
          intent="header"
          label="Fat (g)"
          name="fat"
        />
      </th>
      <th
        key="carbs/.3"
      >
        <Column
          intent="header"
          label="Carbs (g)"
          name="carbs"
        />
      </th>
      <th
        key="protein/.4"
      >
        <Column
          intent="header"
          label="Protein (g)"
          name="protein"
        />
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="1"
    >
      <td
        key=".0"
      >
        <Column
          intent="cell"
          label="Dessert (100g serving)"
          name="name"
          value="Cupcake"
        />
      </td>
      <td
        key=".1"
      >
        <Column
          intent="cell"
          label="Calories"
          name="calories"
          value={305}
        />
      </td>
      <td
        key=".2"
      >
        <Column
          intent="cell"
          label="Fat (g)"
          name="fat"
          value={3.7}
        />
      </td>
      <td
        key=".3"
      >
        <Column
          intent="cell"
          label="Carbs (g)"
          name="carbs"
          value={67}
        />
      </td>
      <td
        key=".4"
      >
        <Column
          intent="cell"
          label="Protein (g)"
          name="protein"
          value={4.3}
        />
      </td>
    </tr>
    <tr
      key="2"
    >
      <td
        key=".0"
      >
        <Column
          intent="cell"
          label="Dessert (100g serving)"
          name="name"
          value="Donut"
        />
      </td>
      <td
        key=".1"
      >
        <Column
          intent="cell"
          label="Calories"
          name="calories"
          value={452}
        />
      </td>
      <td
        key=".2"
      >
        <Column
          intent="cell"
          label="Fat (g)"
          name="fat"
          value={25}
        />
      </td>
      <td
        key=".3"
      >
        <Column
          intent="cell"
          label="Carbs (g)"
          name="carbs"
          value={51}
        />
      </td>
      <td
        key=".4"
      >
        <Column
          intent="cell"
          label="Protein (g)"
          name="protein"
          value={4.9}
        />
      </td>
    </tr>
    <tr
      key="3"
    >
      <td
        key=".0"
      >
        <Column
          intent="cell"
          label="Dessert (100g serving)"
          name="name"
          value="Eclair"
        />
      </td>
      <td
        key=".1"
      >
        <Column
          intent="cell"
          label="Calories"
          name="calories"
          value={262}
        />
      </td>
      <td
        key=".2"
      >
        <Column
          intent="cell"
          label="Fat (g)"
          name="fat"
          value={16}
        />
      </td>
      <td
        key=".3"
      >
        <Column
          intent="cell"
          label="Carbs (g)"
          name="carbs"
          value={24}
        />
      </td>
      <td
        key=".4"
      >
        <Column
          intent="cell"
          label="Protein (g)"
          name="protein"
          value={6}
        />
      </td>
    </tr>
    <tr
      key="5"
    >
      <td
        key=".0"
      >
        <Column
          intent="cell"
          label="Dessert (100g serving)"
          name="name"
          value="Frozen yoghurt"
        />
      </td>
      <td
        key=".1"
      >
        <Column
          intent="cell"
          label="Calories"
          name="calories"
          value={159}
        />
      </td>
      <td
        key=".2"
      >
        <Column
          intent="cell"
          label="Fat (g)"
          name="fat"
          value={6}
        />
      </td>
      <td
        key=".3"
      >
        <Column
          intent="cell"
          label="Carbs (g)"
          name="carbs"
          value={24}
        />
      </td>
      <td
        key=".4"
      >
        <Column
          intent="cell"
          label="Protein (g)"
          name="protein"
          value={4}
        />
      </td>
    </tr>
    <tr
      key="6"
    >
      <td
        key=".0"
      >
        <Column
          intent="cell"
          label="Dessert (100g serving)"
          name="name"
          value="Gingerbread"
        />
      </td>
      <td
        key=".1"
      >
        <Column
          intent="cell"
          label="Calories"
          name="calories"
          value={356}
        />
      </td>
      <td
        key=".2"
      >
        <Column
          intent="cell"
          label="Fat (g)"
          name="fat"
          value={16}
        />
      </td>
      <td
        key=".3"
      >
        <Column
          intent="cell"
          label="Carbs (g)"
          name="carbs"
          value={49}
        />
      </td>
      <td
        key=".4"
      >
        <Column
          intent="cell"
          label="Protein (g)"
          name="protein"
          value={3.9}
        />
      </td>
    </tr>
    <tr
      key="7"
    >
      <td
        key=".0"
      >
        <Column
          intent="cell"
          label="Dessert (100g serving)"
          name="name"
          value="Honeycomb"
        />
      </td>
      <td
        key=".1"
      >
        <Column
          intent="cell"
          label="Calories"
          name="calories"
          value={408}
        />
      </td>
      <td
        key=".2"
      >
        <Column
          intent="cell"
          label="Fat (g)"
          name="fat"
          value={3.2}
        />
      </td>
      <td
        key=".3"
      >
        <Column
          intent="cell"
          label="Carbs (g)"
          name="carbs"
          value={87}
        />
      </td>
      <td
        key=".4"
      >
        <Column
          intent="cell"
          label="Protein (g)"
          name="protein"
          value={6.5}
        />
      </td>
    </tr>
    <tr
      key="8"
    >
      <td
        key=".0"
      >
        <Column
          intent="cell"
          label="Dessert (100g serving)"
          name="name"
          value="Ice cream sandwich"
        />
      </td>
      <td
        key=".1"
      >
        <Column
          intent="cell"
          label="Calories"
          name="calories"
          value={237}
        />
      </td>
      <td
        key=".2"
      >
        <Column
          intent="cell"
          label="Fat (g)"
          name="fat"
          value={9}
        />
      </td>
      <td
        key=".3"
      >
        <Column
          intent="cell"
          label="Carbs (g)"
          name="carbs"
          value={37}
        />
      </td>
      <td
        key=".4"
      >
        <Column
          intent="cell"
          label="Protein (g)"
          name="protein"
          value={4.3}
        />
      </td>
    </tr>
    <tr
      key="9"
    >
      <td
        key=".0"
      >
        <Column
          intent="cell"
          label="Dessert (100g serving)"
          name="name"
          value="Jelly Bean"
        />
      </td>
      <td
        key=".1"
      >
        <Column
          intent="cell"
          label="Calories"
          name="calories"
          value={375}
        />
      </td>
      <td
        key=".2"
      >
        <Column
          intent="cell"
          label="Fat (g)"
          name="fat"
          value={0}
        />
      </td>
      <td
        key=".3"
      >
        <Column
          intent="cell"
          label="Carbs (g)"
          name="carbs"
          value={94}
        />
      </td>
      <td
        key=".4"
      >
        <Column
          intent="cell"
          label="Protein (g)"
          name="protein"
          value={0}
        />
      </td>
    </tr>
    <tr
      key="10"
    >
      <td
        key=".0"
      >
        <Column
          intent="cell"
          label="Dessert (100g serving)"
          name="name"
          value="KitKat"
        />
      </td>
      <td
        key=".1"
      >
        <Column
          intent="cell"
          label="Calories"
          name="calories"
          value={518}
        />
      </td>
      <td
        key=".2"
      >
        <Column
          intent="cell"
          label="Fat (g)"
          name="fat"
          value={26}
        />
      </td>
      <td
        key=".3"
      >
        <Column
          intent="cell"
          label="Carbs (g)"
          name="carbs"
          value={65}
        />
      </td>
      <td
        key=".4"
      >
        <Column
          intent="cell"
          label="Protein (g)"
          name="protein"
          value={7}
        />
      </td>
    </tr>
    <tr
      key="11"
    >
      <td
        key=".0"
      >
        <Column
          intent="cell"
          label="Dessert (100g serving)"
          name="name"
          value="Lollipop"
        />
      </td>
      <td
        key=".1"
      >
        <Column
          intent="cell"
          label="Calories"
          name="calories"
          value={392}
        />
      </td>
      <td
        key=".2"
      >
        <Column
          intent="cell"
          label="Fat (g)"
          name="fat"
          value={0.2}
        />
      </td>
      <td
        key=".3"
      >
        <Column
          intent="cell"
          label="Carbs (g)"
          name="carbs"
          value={98}
        />
      </td>
      <td
        key=".4"
      >
        <Column
          intent="cell"
          label="Protein (g)"
          name="protein"
          value={0}
        />
      </td>
    </tr>
    <tr
      key="12"
    >
      <td
        key=".0"
      >
        <Column
          intent="cell"
          label="Dessert (100g serving)"
          name="name"
          value="Marshmallow"
        />
      </td>
      <td
        key=".1"
      >
        <Column
          intent="cell"
          label="Calories"
          name="calories"
          value={318}
        />
      </td>
      <td
        key=".2"
      >
        <Column
          intent="cell"
          label="Fat (g)"
          name="fat"
          value={0}
        />
      </td>
      <td
        key=".3"
      >
        <Column
          intent="cell"
          label="Carbs (g)"
          name="carbs"
          value={81}
        />
      </td>
      <td
        key=".4"
      >
        <Column
          intent="cell"
          label="Protein (g)"
          name="protein"
          value={2}
        />
      </td>
    </tr>
    <tr
      key="13"
    >
      <td
        key=".0"
      >
        <Column
          intent="cell"
          label="Dessert (100g serving)"
          name="name"
          value="Nougat"
        />
      </td>
      <td
        key=".1"
      >
        <Column
          intent="cell"
          label="Calories"
          name="calories"
          value={360}
        />
      </td>
      <td
        key=".2"
      >
        <Column
          intent="cell"
          label="Fat (g)"
          name="fat"
          value={19}
        />
      </td>
      <td
        key=".3"
      >
        <Column
          intent="cell"
          label="Carbs (g)"
          name="carbs"
          value={9}
        />
      </td>
      <td
        key=".4"
      >
        <Column
          intent="cell"
          label="Protein (g)"
          name="protein"
          value={37}
        />
      </td>
    </tr>
    <tr
      key="14"
    >
      <td
        key=".0"
      >
        <Column
          intent="cell"
          label="Dessert (100g serving)"
          name="name"
          value="Oreo"
        />
      </td>
      <td
        key=".1"
      >
        <Column
          intent="cell"
          label="Calories"
          name="calories"
          value={437}
        />
      </td>
      <td
        key=".2"
      >
        <Column
          intent="cell"
          label="Fat (g)"
          name="fat"
          value={18}
        />
      </td>
      <td
        key=".3"
      >
        <Column
          intent="cell"
          label="Carbs (g)"
          name="carbs"
          value={63}
        />
      </td>
      <td
        key=".4"
      >
        <Column
          intent="cell"
          label="Protein (g)"
          name="protein"
          value={4}
        />
      </td>
    </tr>
  </tbody>
</table>
`;
